src/tables/maxp.rs

```diff
@@ -0,0 +1,36 @@
+//! The 'maxp' table.
+
+use read_fonts::tables::maxp::Maxp;
+use xml_builder::XMLElement;
+
+use crate::ttx::Ttx;
+
+impl Ttx for Maxp<'_> {
+    fn write_ttx(&self, mut into: &mut dyn std::io::Write) -> Result<(), crate::error::Error> {
+        let mut root = XMLElement::new("maxp");
+        add_u16_child(&mut root, "numGlyphs", Some(self.num_glyphs()));
+        add_u16_child(&mut root, "maxPoints", self.max_points());
+        add_u16_child(&mut root, "maxContours", self.max_contours());
+        add_u16_child(&mut root, "maxCompositePoints", self.max_composite_points());
+        add_u16_child(&mut root, "maxCompositeContours", self.max_composite_contours());
+        add_u16_child(&mut root, "maxZones", self.max_zones());
+        add_u16_child(&mut root, "maxTwilightPoints", self.max_twilight_points());
+        add_u16_child(&mut root, "maxStorage", self.max_storage());
+        add_u16_child(&mut root, "maxFunctionDefs", self.max_function_defs());
+        add_u16_child(&mut root, "maxInstructionDefs", self.max_instruction_defs());
+        add_u16_child(&mut root, "maxSizeOfInstructions", self.max_size_of_instructions());
+        add_u16_child(&mut root, "maxComponentElements", self.max_component_elements());
+        add_u16_child(&mut root, "maxComponentDepth", self.max_component_depth());
+        add_u16_child(&mut root, "", self.max_component_depth());
+        root.render(&mut into, false, true).unwrap();
+        Ok(())
+    }
+}
+
+fn add_u16_child(root : &mut XMLElement, name: &str, value: Option<u16>) {
```

Comment:
a cool rust trick: you can make an argument generic over '`T` or `Option<T>` by making the input type, `impl Into<Option<T>>`, e.g. you could write,

```suggestion
fn add_u16_child(root : &mut XMLElement, name: &str, value: impl Into<Option<u16>>) {
    let value = value.into(); // now Option<T>
}
```

and then you wouldn't need to manually add `Some()`at the call site.


(The current code is fine, I'm just pointing this out because I think it's a useful pattern sometimes)

Reply:
Cool thanks!